src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,198 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V>[] currentTable = table;
+        int hash = hash(key);
+        if (currentTable == null || currentTable.length == 0) {
+            currentTable = resize();
+        }
+        if (currentTable[hash % currentTable.length] == null) {
+            currentTable[hash % currentTable.length] = new Node<>(hash, key, value, null);
+        } else {
+            Node<K, V> currentNode = currentTable[hash % currentTable.length];
+            if ((currentNode.key == null) && (key == null)) { // если обе нулл
+                currentNode.value = value;
+                size--;
+            } else {
+                if ((currentNode.hash == hash)
+                        && ((currentNode.key == key)
+                        || (currentNode.key != null && currentNode.key.equals(key)))) {
+                    currentNode.value = value;
+                    size--;
+                } else if (currentNode.next != null) {
+                    while (currentNode.next != null) {
+                        currentNode = currentNode.next;
+                        if ((currentNode.key == null) && (key == null)) {
+                            currentNode.value = value;
+                            size--;
+                            break;
+                        } else {
+                            if ((currentNode.hash == hash)
+                                    && ((currentNode.key == key)
+                                    || (currentNode.key != null && currentNode.key.equals(key)))) {
+                                currentNode.value = value;
+                                size--;
+                                break;
+                            }
+                        }
+                    }
+                    if (currentNode.next == null) {
+                        currentNode.next = new Node<>(hash, key, value, null);
+                    }
+                } else { //esli currentNode.next == 0
+                    currentNode.next = new Node<>(hash, key, value, null);
+                }
+            }
+        }
+        size++;
+        if (size >= threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> element;
+        if (size == 0) {
+            return null;
+        }
+        for (int i = 0; i < table.length; i++) {
```

Comment:
```suggestion
         @Override
    public V getValue(K key) {
        if (table[getBucket(key)] == null) {
            return null;
        } else {
            Node<K, V> firstNode = table[getBucket(key)];
            while (firstNode != null) {
                if (firstNode.key == key || firstNode.key != null && firstNode.key.equals(key)) {
                    return firstNode.value;
                }
                firstNode = firstNode.next;
            }
        }
        return null;
    }
    
    private int getBucket(K key) {
        if (key == null) {
            return 0;
        }
        return Math.abs(key.hashCode() % table.length);
    }
```
Totaly wrong algorithm, you don't have to check each backet! In this case your map will work the same like LinkedList by moving from one Node to next. You have to count which backet you need by (hashCode % currentCapacity) ant check only this backet in array. In your solushion you can't get extra performance. My code is an example.